### ISSUE TITLE
net/http: recommend a polyfill for node.js

### ISSUE THREAD

**Issue by broady** [gopherjs/gopherjs#518]
`xmlhttprequest` doesn't work very well:
https://www.npmjs.com/package/xmlhttprequest

For some reason the first thing it does when it gets a request is abort it. wtf?!?!?

``` js
  this.open = function(method, url, async, user, password) {
    this.abort();
    errorFlag = false;
```

**Comment by shurcooL** [comment]
If there's an overall better replacement for `xmlhttprequest`, we should be able to update to it. We'll take PRs. :)

**Comment by neelance** [comment]
Note: GopherJS's support for Node.js is only intended for package testing. For every normal workload it should be preferable to simply use normal Go.

No suggestions on a polyfill, sorry.

**Comment by MartinSahlen** [comment]
@broady You are not alone! I have tried an erred somewhat with this. Background: I am currently working on porting google cloud functions to many different languages, including clojure, f# and now golang:
- https://github.com/MartinSahlen/go-cloud-fn
- https://github.com/MartinSahlen/fsharp-gcloud-functions
- https://github.com/MartinSahlen/cloud-fn-test

Back to the matter at hand: I tried 

- https://github.com/matthew-andrews/isomorphic-fetch + https://github.com/stefanpenner/es6-promise
- https://github.com/bitinn/node-fetch

For the fetch API. I also had to include https://www.npmjs.com/package/web-streams-polyfill to satisfy https://github.com/gopherjs/gopherjs/blob/f2fc182abb28925c7aa40502db839c6ebd7f7171/compiler/natives/src/net/http/http.go#L18 demand for this to be present. However, none of this worked. so I tried https://github.com/driverdan/node-XMLHttpRequest. However, this just aborted and caused and endless loop of retries.

I did have a lot more success with https://github.com/pwnall/node-xhr2 that worked and seems to be stable so far. My approach was to use create a file called polyfill.js (https://github.com/MartinSahlen/go-cloud-fn/blob/master/polyfill/polyfill.js) that imported the modules i needed and then use https://github.com/webpack/webpack to bundle all dependences into a standalone file (gopherjs doesnt handle this for us it seems, and no hard feelings, there are other tools doing that already). 

Final notes: So, I know gopherjs isnt meant to do this but at least this shows that it certainly is able to do it. The reason for me wanting this that is that google cloud functions currently are basically just handlers for the node express framework (and I love go ❤️ ). Maybe at some point it will work with node.js. I will be happy to discuss this further and contribute if there is a need and wish for it!

**Comment by shurcooL** [comment]
Even though this issue was closed, a similar issue #586 was later opened and there's some additional discussion/searches for a good polyfill there.

Let's continue any discussion in the open issue #586, but I just wanted to point that out for anyone following this issue, and for posterity.
